counsel.el

```diff
@@ -5096,6 +5100,44 @@ build environments.")
                      (or (buffer-file-name) default-directory)
                      ".git")))
 
+;; This is loosely based on the bash completion code
+(defun counsel--get-make-calls (root)
+  "Return a list of make calls for a given directory.
+
+We search the Makefile for a list of PHONY targets which are generally
+the top-level targets a make system provides."
+  (let ((default-directory root))
+    (mapcar
+     (lambda(target)
+       (format "cd %s; make %s %s" root counsel-compile-make-opts target))
```

Comment:
Better `cd %s && make ...`?

Reply:
Why not `make -C %s ...`?

Reply:
I think I originally had cd %s && make but it broke when I chsh'd to fish whereas ; works with both - however make -C might be a better bet (use less space).

Reply:
Keep in mind that it also changes semantics of the build because `PWD` inside `make` will be different in both cases.

Reply:
I think the use should decide `PWD` of execution and it should be the `default-directory`.  That is `cd` solution is not viable.  However, just relying on plain `default-directory` is not useful as well.  That is, when I'm calling `counsel-compile` from some buffer, I don't want the `default-directory` of it to become potentially polluted by the build.  So the `let`-binding of `default-directory` per `counsel-compile` invocation should be configurable and customizable by user based on the project.  Keep in mind that `make -C` should still be used to act on the right `Makefile` anyway.